metrics/registry.lua

```diff
@@ -70,6 +70,14 @@ function Registry:register_callback(callback)
     end
 end
 
+function Registry:unregister_callback(callback)
+    for i, registered_callback in ipairs(self.callbacks) do
+        if registered_callback == callback then
```

Comment:
Isn't it a good time to make it a hash table like metrics registry?

Reply:
But since we don't have any explicit keys here, it is not so obvious how to organize it

Reply:
+1

Reply:
#276